lunes_cms/help/templates/public_upload.html

```diff
@@ -65,28 +65,46 @@
         3: "das",
         4: "die (Plural)"
       }
+      let buffer_array = []
 
-      $( document ).ready(function() {
+      function Comparator(a, b) {
+        if (a[1] < b[1]) return -1;
+        if (a[1] > b[1]) return 1;
+        return 0;
+      }
 
+      $( document ).ready(function() {
+        $("#inputDiscipline").append(new Option("Alle Berufe", 0))
         lunes_disciplines.forEach(function(element) {
           if( element[1] !== "Allgemeine Grundlagen") {
             $("#inputDiscipline").append(new Option(element[1], element[0]))
           }
         });
-
         $("#inputDiscipline").change(function() {
           var discipline_id = $("#inputDiscipline").val();
-          $("#inputDocument").find('option').remove();
-          lunes_training_sets.forEach(function(element) {
-            if(lunes_disc_sets_map[discipline_id].includes(element[0])) {
-              var training_set_id = element[0];
-              lunes_documents.forEach(function(element) {
-                if(element[3] == training_set_id) {
-                  $("#inputDocument").append(new Option( article_map[element[2]] + " " + element[1], element[0]));
-                }
-              });
-            }
-          });
+          if (discipline_id != 0) {
```

Comment:
```suggestion
          if (discipline_id == "all") {
            buffer_array = lunes_documents;
          } else if (discipline_id != "none") {
```
I think I would reorder the loop to make sure we don't throw a console error when no discipline is selected.